k8s-scheduler/src/main/java/com/vmware/dcm/ScopedModel.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2018-2020 VMware, Inc. All Rights Reserved.
+ *
+ * SPDX-License-Identifier: BSD-2
+ */
+
+package com.vmware.dcm;
+
+import com.vmware.dcm.k8s.generated.Tables;
+import org.jooq.Condition;
+import org.jooq.DSLContext;
+import org.jooq.Record;
+import org.jooq.Result;
+import org.jooq.SortField;
+import org.jooq.Table;
+import static org.jooq.impl.DSL.field;
+import static org.jooq.impl.DSL.selectFrom;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+/**
+ * A Scope to reduce the Model's solver input.
+ */
+
+public class ScopedModel {
+    private static final Logger LOG = LoggerFactory.getLogger(ScopedModel.class);
+    private final DSLContext conn;
+    private final Model model;
+
+    // dynamically adjuct number of scoped nodes with limitTune
+    private double limitTune = 1.0;
+    // TODO: add extra limitPriority (provided as a hint)
+    //       divide limitTune to (three) categories according to affinity
+    //       actually tune limitTune
+
+    // TODO: add testcase
+    private double sortWeightCpu = 0.8;
+    private double sortWeightMemory = 0.2;
+
+    // TODO: capture successful schedules history
+
+
+    ScopedModel(final DSLContext conn, final Model model) {
+        this.conn = conn;
+        this.model = model;
+    }
+
+    private Collection<Condition> getWhereClause(final Set<String> nodeSet) {
+        final Collection<Condition> conditions = new ArrayList<>();
+        conditions.add(Tables.SPARE_CAPACITY_PER_NODE.NAME.in(nodeSet));
+
+        return conditions;
+    }
+
+    private SortField<?> getSorting() {
+        return field(sortWeightCpu + " * cpu_remaining + "
+                + sortWeightMemory + " * memory_remaining").desc();
+    }
+
+    private int getLimit(final int cnt) {
```

Comment:
What's cnt? What's limitTune? With methods like this, you want to leave a comment above the method explaining why you do this, and use better variable names so that the code is easy to read.